source/material_model/visco_plastic.cc

```diff
@@ -180,14 +191,32 @@ namespace aspect
           // to compute the elastic force term.
           bool plastic_yielding = false;
           IsostrainViscosities isostrain_viscosities;
+
           if (in.requests_property(MaterialProperties::viscosity) || in.requests_property(MaterialProperties::additional_outputs))
             {
               // Currently, the viscosities for each of the compositional fields are calculated assuming
               // isostrain amongst all compositions, allowing calculation of the viscosity ratio.
               // TODO: This is only consistent with viscosity averaging if the arithmetic averaging
               // scheme is chosen. It would be useful to have a function to calculate isostress viscosities.
-              isostrain_viscosities =
-                rheology->calculate_isostrain_viscosities(in, i, volume_fractions, phase_function_values, n_phase_transitions_for_each_chemical_composition);
+              // Methods for averaging among phases depend on whether the most dominant phases are looked up
+              // for each composition in its respective lookup table. A separate phase function class
+              // manages the averaging if the user chooses this option.
+              if (use_dominant_phase_for_viscosity)
+                {
+                  std::vector<double> phase_function_discrete_values(phase_function_discrete.n_phase_transitions(), 0.0);
```

Comment:
I have made this change, but as you can see, the line I moved to 114 incorporates a bit of complexity

```
std::vector<double> phase_function_discrete_values = (use_dominant_phase_for_viscosity?
                                                            std::vector<double>(phase_function_discrete->n_phase_transitions(), 0.0): std::vector<double>());

```
The reason for this is that we make the phase_function_discrete a unique_ptr, so that when the use_dominant_phase_for_viscosity is False, the pointer is not initiated yet. On the other hand, when I did this implementation I did feel what you meant in the other comments for the unique_ptr, that it forced me to be more careful.